homework/index.js

```diff
@@ -1,46 +1,165 @@
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderRepositories(repository) {
+    const leftBox = document.getElementById('left-box');
+    leftBox.innerHTML = '';
+    const rTable = createAndAppend('table', leftBox);
+    const tbody = createAndAppend('tbody', rTable, {
+      class: 'table_elms',
+    });
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository      :',
+    });
+    createAndAppend('td', tr1, {
+      text: repository.name,
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, {
+      text: 'Description    :',
+    });
+    createAndAppend('td', tr2, {
+      text: repository.description,
+    });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, {
+      text: 'Forks          :',
+    });
+    createAndAppend('td', tr3, {
+      text: repository.forks,
+    });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, {
+      text: 'Updated        :',
+    });
+    createAndAppend('td', tr4, {
+      text: new Date(repository.updated_at).toLocaleString('en-GB'),
+    });
+  }
+
+  function renderContributions(repository) {
+    const rightBox = document.getElementById('right-box');
+    rightBox.innerHTML = '';
+    fetchJSON(repository.contributors_url, (error, contributions) => {
+      if (error) {
+        renderError(error);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      createAndAppend('h2', rightBox, {
+        text: 'Contributions',
+        class: 'contrib-title',
+      });
+      const contributionsList = createAndAppend('ul', rightBox, {
+        class: 'list-container',
+      });
+      contributions.forEach(contribution => {
+        const listItem = createAndAppend('li', contributionsList, {
+          class: 'contrib',
+        });
+        createAndAppend('img', listItem, {
+          class: 'contrib_img',
+          src: contribution.avatar_url,
+        });
+        createAndAppend('span', listItem, {
+          class: 'contrib-name',
+          text: contribution.login,
+        });
+        createAndAppend('span', listItem, {
+          class: 'contrib-count',
+          text: contribution.contributions,
+        });
+      });
     });
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      id: 'header',
+    });
+    createAndAppend('h1', header, {
+      class: 'header-title',
+      text: 'HYF Repositories',
+    });
+    const select = createAndAppend('select', header);
+    const container = createAndAppend('container', root, {
```

Comment:
`'container'` is not the name of a valid HTML element. Since you have used `section` as child element a `main` semantic HTML tag seems appropriate here.

Reply:
I used main.